impeller/renderer/backend/vulkan/allocator_vk.cc

```diff
@@ -15,6 +15,43 @@
 
 namespace impeller {
 
+// Maximum size to use VMA image suballocation. Any allocation greater than or
+// equal to this value will use a dedicated VkDeviceMemory.
+constexpr size_t kImageSizeThresholdForDedicatedMemoryAllocation =
+    4 * 1024 * 1024;
+
+static bool CreateBufferPool(VmaAllocator allocator, VmaPool* pool) {
+  vk::BufferCreateInfo buffer_info;
+  buffer_info.usage = vk::BufferUsageFlagBits::eVertexBuffer |
+                      vk::BufferUsageFlagBits::eIndexBuffer |
+                      vk::BufferUsageFlagBits::eUniformBuffer |
+                      vk::BufferUsageFlagBits::eStorageBuffer |
+                      vk::BufferUsageFlagBits::eTransferSrc |
+                      vk::BufferUsageFlagBits::eTransferDst;
+  buffer_info.size = 1u;  // doesn't matter
+  buffer_info.sharingMode = vk::SharingMode::eExclusive;
+  auto buffer_info_native =
+      static_cast<vk::BufferCreateInfo::NativeType>(buffer_info);
+
+  VmaAllocationCreateInfo sampleAllocCreateInfo = {};
+  sampleAllocCreateInfo.usage = VMA_MEMORY_USAGE_AUTO;
+
+  uint32_t memTypeIndex;
+  VkResult res = vmaFindMemoryTypeIndexForBufferInfo(
+      allocator, &buffer_info_native, &sampleAllocCreateInfo, &memTypeIndex);
+
+  VmaPoolCreateInfo poolCreateInfo = {};
+  poolCreateInfo.memoryTypeIndex = memTypeIndex;
+  poolCreateInfo.blockSize = 128ull * 1024 * 1024;
```

Comment:
As mentioned in https://github.com/flutter/engine/pull/43313#discussion_r1246168700, let's put the tunables in a separate file so everything is available at a glance?